cmd/ci-secret-bootstrap/main.go

```diff
@@ -471,7 +471,14 @@ func fetchUserSecrets(secretsMap map[string]map[types.NamespacedName]coreapi.Sec
 	var errs []error
 	for secretName, secretKeys := range userSecrets {
 		logger := logrus.WithField("secret", secretName.String())
+		allowedClusters := sets.String{}
+		if secretKeys[vaultapi.SecretSyncTargetClusterKey] != "" {
+			allowedClusters.Insert(strings.Split(secretKeys[vaultapi.SecretSyncTargetClusterKey], ",")...)
+		}
 		for _, cluster := range targetClusters {
+			if len(allowedClusters) != 0 && !allowedClusters.Has(cluster) {
+				continue
+			}
```

Comment:
We could remove this check on every iteration by looping over either the original cluster list or the restricted one.

```go
clusterSet := sets.NewString(targetClusters...)
// …
selectedClusters := targetClusters
if v := secretKeys[vaultapi.SecretSyncTargetClusterKey]; v != "" {
	selectedClusters = clusterSet.Intersection(sets.NewString(strings.Split(v, ",")...)).List()
}
for _, cluster := range selectedClusters {
```

Reply:
Updated